File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/base/BasePresenter.java
package com.kin.ecosystem.base;

import android.support.annotation.CallSuper;

public class BasePresenter<T extends IBaseView> implements IBasePresenter<T> {

    protected T view;

    @CallSuper
    @Override
    public void onAttach(T view) {
        this.view = view;

    }

    @CallSuper
    @Override
    public void onDetach() {
        view = null;
    }
}

File: sdk/src/main/java/com/kin/ecosystem/base/BasePresenter.kt
package com.kin.ecosystem.base

import android.support.annotation.CallSuper

open class BasePresenter<T : IBaseView> : IBasePresenter<T> {

    var view: T? = null
        protected set

    @CallSuper
    override fun onAttach(view: T) {
        this.view = view

    }

    @CallSuper
    override fun onDetach() {
        view = null
    }
}
